pkg/cloudprovider/kubevirt/loadbalancer.go

```diff
@@ -167,9 +173,9 @@ func (lb *loadbalancer) EnsureLoadBalancerDeleted(ctx context.Context, clusterNa
 	return nil
 }
 
-func (lb *loadbalancer) getLoadBalancerService(lbName string) (*corev1.Service, bool, error) {
-	service, err := lb.kubevirt.CoreV1().Services(lb.namespace).Get(lbName, metav1.GetOptions{})
-	if err != nil {
+func (lb *loadbalancer) getLoadBalancerService(ctx context.Context, lbName string) (*corev1.Service, bool, error) {
+	var service *corev1.Service
+	if err := lb.client.Get(ctx, client.ObjectKey{Name: lbName, Namespace: lb.namespace}, service); err != nil {
```

Comment:
You can't pass a nil pointer here,
```suggestion
	var service corev1.Service
	if err := lb.client.Get(ctx, client.ObjectKey{Name: lbName, Namespace: lb.namespace}, &service); err != nil {
```